src/components/ModalLedgerSignToken.js

```diff
@@ -0,0 +1,281 @@
+    canSend(false);
+    setOk(false);
+    setError(null);
+    // send token data to ledger
+    blockClose(true);
+    ledger.signToken(token);
+  }
+
+  /**
+   * Handle the response of a send token data call to Ledger.
+   * Will list all tokens that failed verification, if all pass then start sign tx
+   *
+   * @param {IpcRendererEvent} event May be used to reply to the event
+   * @param {Object} arg Data returned from the send token data call
+   */
+  const handleSignature = (event, arg) => {
+    console.log('handleSignature', arg);
+    blockClose(false);
+    if (arg.success) {
+      // save token signature on storage
+      tokens.addTokenSignature(token.uid, arg.data.toString("hex"));
+      setOk(true);
+    } else {
+      // TODO: handle error (like user deny)
+      console.log('user_deny', arg.error.status == 0x6985);
+      switch (arg.error.status) {
+        // user deny
+        case 0x6985: 
+          setError('user_deny');
+          break;
+        case 0xb00a: 
+          setError('invalid_token');
+          break;
+        default:
+          setError('unknown_error');
+      }
+    }
+  }
+
+  /**
+   * Handle the response of a send token data call to Ledger.
+   * Will list all tokens that failed verification, if all pass then start sign tx
+   *
+   * @param {IpcRendererEvent} event May be used to reply to the event
+   * @param {Object} arg Data returned from the send token data call
+   */
+  const handleVerify = (event, arg) => {
+    console.log('handleVerify', arg);
+    blockClose(false);
+    if (arg.success) {
+      setOk(true);
+    } else {
+      // Invalid signature, we should start the signing process
+      canSend(true);
+    }
+  }
+
+  /**
+   * Effects
+   */
+
+  // useEffect second parameter [] ensures this runs only once
+  useEffect(() => {
+    $(`#${modalId}`).on('hide.bs.modal', (e) => {
+      // clean state when closing
+      initialState();
+      // remove ledger listeners when closing
+      if (IPC_RENDERER) {
+        IPC_RENDERER.removeAllListeners("ledger:tokenSignature");
+        IPC_RENDERER.removeAllListeners("ledger:tokenSignatureValid");
+      }
+    });
+
+    $(`#${modalId}`).on('shown.bs.modal', (e) => {
+      // ledger listeners
+      if (IPC_RENDERER) {
+        IPC_RENDERER.on("ledger:tokenSignature", handleSignature);
+        IPC_RENDERER.on("ledger:tokenSignatureValid", handleVerify);
+      }
+      // verify token information when opening modal
+      verifyToken();
+    });
+
+    return () => {
+      // remove all listeners on unmount
+      $(`#${modalId}`).off();
+    }
+  }, []);
+
+  // token change -> go to begin and verify it
+  useEffect(() => {
+    $(`#${modalId}`).modal('hide');
+    // initialState();
+    // verify new token
+    // verifyToken();
+  }, [token]);
+
+  /**
+   * Render
+   */
+
+  const renderToken = () => {
+    // TODO: add a backgroung color or contour to token info
+    return (
+      <div id="show-token-info">
```

Comment:
What's this id used for?

Reply:
It had meaning on a previous iteration, but this is not used anymore, removing it.